### ISSUE TITLE
Native nuget package

### ISSUE THREAD

**Issue by sandcastle** [kleisauke/net-vips#21]
Currently, the net-vips Nuget contains just windows x86/x64, which does not help me - I'm developing on Mac and deploying to Debian (which has an outdated vips version).

It would be great if net-vips followed a similar strategy to SkiaSharp, where they now have a new Native Nuget that contains the target runtimes for win, osx, linux, etc.

https://github.com/mono/SkiaSharp/wiki/SkiaSharp-Native-Assets-for-Linux

The skia linux nuget:
https://www.nuget.org/packages/SkiaSharp.NativeAssets.Linux

What are your thoughts about adding something like this to the project?

**Comment by kleisauke** [comment]
Things that should be done before we could declare 1.1:
- [ ] Separate bundled libvips x86/x64 Windows binary to `NetVips.Native.win-x86` and `NetVips.Native.win-x64` NuGet packages.
   - [ ] Should be placed in the architecture-specific folders (`runtimes/win-x86/native/` and `runtimes/win-x64/native/`).
   - [ ] Remove `NetVips.targets` (will drop support for `UseGlobalLibvips`).
- [ ] Add pre-compiled Linux x64 glibc binaries (Ubuntu, Debian, etc.).
NuGet package name: `NetVips.Native.linux-x64`.
Runtime folder: `runtimes/linux-x64/native/`.
- [ ] Add pre-compiled Linux x64 musl binaries (Alpine Linux).
NuGet package name: `NetVips.Native.linuxmusl-x64`.
Runtime folder: `runtimes/linux-musl-x64/native/`.
- [ ] Add pre-compiled OS X x64 binaries.
NuGet package name: `NetVips.Native.darwin-x64` (or `NetVips.Native.osx-x64` (?)).
Runtime folder: `runtimes/osx-x64/native/`.
- [ ] Should we use the version number of libvips for these packages?

Wishlist:
- [ ] Add pre-compiled Linux ARMv6 binaries (Raspberry Pi 1).
NuGet package name: `NetVips.Native.linux-armv6`.
Runtime folder: `runtimes/linux-arm/native/`.
- [ ] Add pre-compiled Linux ARMv7-A binaries (Raspberry Pi 2 Model B).
NuGet package name: `NetVips.Native.linux-armv7`.
Runtime folder: `runtimes/linux-arm/native/`.
- [ ] Add pre-compiled Linux ARMv8-A binaries (Raspberry Pi 3 Model B+).
NuGet package name: `NetVips.Native.linux-armv8`.
Runtime folder: `runtimes/linux-arm/native/`.

**Comment by kleisauke** [comment]
I've created some NuGet packages (commit https://github.com/kleisauke/net-vips/commit/5ace21fb0877f3e2035374dd46dadd7c32aa767e) that contains the pre-compiled libvips binaries for a few distros (see [this repo](https://github.com/kleisauke/libvips-packaging) for details):

|                    |NuGet Package¹|
|--------------------|:------------:|
|**Windows 64-bit**  |[![NetVips.Native.win-x64](https://img.shields.io/nuget/v/NetVips.Native.win-x64.svg)](https://www.nuget.org/packages/NetVips.Native.win-x64)|
|**Windows 32-bit**  |[![NetVips.Native.win-x64](https://img.shields.io/nuget/v/NetVips.Native.win-x86.svg)](https://www.nuget.org/packages/NetVips.Native.win-x86)|
|**Linux x64 glibc²**|[![NetVips.Native.linux-x64](https://img.shields.io/nuget/v/NetVips.Native.linux-x64.svg)](https://www.nuget.org/packages/NetVips.Native.linux-x64)|
|**Linux x64 musl³** |[![NetVips.Native.linux-musl-x64](https://img.shields.io/nuget/v/NetVips.Native.linux-musl-x64.svg)](https://www.nuget.org/packages/NetVips.Native.linux-musl-x64)|
|**macOS x64**       |[![NetVips.Native.osx-x64](https://img.shields.io/nuget/v/NetVips.Native.osx-x64.svg)](https://www.nuget.org/packages/NetVips.Native.osx-x64)|

¹ The version number of these NuGet packages is in sync with libvips' version number.  
² Uses glibc as the standard C library (Ubuntu, Debian, etc).  
³ Uses musl as the standard C library (Alpine, Gentoo Linux, etc).

The bundled libvips Windows binaries were split into different NuGet packages. Is there anyone willing to test this? Especially on Alpine and macOS (since I tested these distros only on Travis and Docker).

For testing these packages, you could run the vips test suite (requires .NET Core >= 2.2):
```bash
git clone -b 1.1.0 --single-branch https://github.com/kleisauke/net-vips
cd net-vips

# For OS X
dotnet add tests/NetVips.Tests package NetVips.Native.osx-x64 -v 8.7.4-beta1

# For Linux x64 glibc
dotnet add tests/NetVips.Tests package NetVips.Native.linux-x64 -v 8.7.4-beta1

# For Linux x64 musl
dotnet add tests/NetVips.Tests package NetVips.Native.linux-musl-x64 -v 8.7.4-beta1

# For Alpine 3.8+ (see https://github.com/dotnet/corefx/issues/34316)
export DOTNET_RUNTIME_ID=linux-musl-x64

./build.sh --target=Test
```

**Comment by sandcastle** [comment]
@kleisauke - nice work on getting this going!

I will do some testing on Mac, Debian and Alpine (when I get some time in the next week).

**Comment by sandcastle** [comment]
@kleisauke - I'm seeing the following error when trying to run on alpine 3.8/3.9.

I'm not familiar with cake; have you seen this before?

```sh
/app/tools/dotnet-cake: line 1: syntax error: unterminated quoted string
```

**Comment by sandcastle** [comment]
# Test Results

## ✅ OSX

_Configuration:_  
- .Net Core SDK - 2.2.104
- .Net Core Runtime - 2.2.2
- OSX - 10.14.3


_On the computer:_  
```sh
./build.sh --target=Test
```

_Test results:_  
```sh
Installing Cake 0.32.1...
You can invoke the tool using the following command: dotnet-cake
Tool 'cake.tool' (version '0.32.1') was successfully installed.

========================================
Clean
========================================

========================================
Restore
========================================
  Restoring packages for /Users/glenn/dev/net-vips/src/NetVips/NetVips.csproj...
  Installing Fody 4.2.1.
  Installing ModuleInit.Fody 1.9.0.
  Generating MSBuild file /Users/glenn/dev/net-vips/src/NetVips/obj/NetVips.csproj.nuget.g.props.
  Generating MSBuild file /Users/glenn/dev/net-vips/src/NetVips/obj/NetVips.csproj.nuget.g.targets.
  Restore completed in 6.17 sec for /Users/glenn/dev/net-vips/src/NetVips/NetVips.csproj.

========================================
Build
========================================
Microsoft (R) Build Engine version 15.9.20+g88f5fadfbe for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  NetVips -> /Users/glenn/dev/net-vips/src/NetVips/bin/AnyCPU/Release/netstandard2.0/OSX/NetVips.dll

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:03.56

========================================
Test
========================================
Build started, please wait...
Build completed.

Test run for /Users/glenn/dev/net-vips/tests/NetVips.Tests/bin/Release/netcoreapp2.2/NetVips.Tests.dll(.NETCoreApp,Version=v2.2)
Microsoft (R) Test Execution Command Line Tool Version 15.9.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
[xUnit.net 00:00:01.48]     NetVips.Tests.ForeignTests.TestOpenexrLoad [SKIP]
Skipped  NetVips.Tests.ForeignTests.TestOpenexrLoad
[xUnit.net 00:00:02.99]     NetVips.Tests.ForeignTests.TestPdfLoad [SKIP]
Skipped  NetVips.Tests.ForeignTests.TestPdfLoad
[xUnit.net 00:00:04.56]     NetVips.Tests.CreateTests.TestFractsurf [SKIP]
Skipped  NetVips.Tests.CreateTests.TestFractsurf
[xUnit.net 00:00:05.79]     NetVips.Tests.ForeignTests.TestOpenslideLoad [SKIP]
Skipped  NetVips.Tests.ForeignTests.TestOpenslideLoad
[xUnit.net 00:00:09.41]     NetVips.Tests.ForeignTests.TestMatLoad [SKIP]
Skipped  NetVips.Tests.ForeignTests.TestMatLoad
[xUnit.net 00:00:10.32]     NetVips.Tests.ForeignTests.TestsNiftiLoad [SKIP]
[xUnit.net 00:00:10.35]     NetVips.Tests.ForeignTests.TestsFitsLoad [SKIP]
Skipped  NetVips.Tests.ForeignTests.TestsNiftiLoad
Skipped  NetVips.Tests.ForeignTests.TestsFitsLoad
[xUnit.net 00:00:13.37]     NetVips.Tests.ForeignTests.TestMagickLoad [SKIP]
Skipped  NetVips.Tests.ForeignTests.TestMagickLoad
[xUnit.net 00:00:23.26]     NetVips.Tests.ConvolutionTests.TestConva [SKIP]
Skipped  NetVips.Tests.ConvolutionTests.TestConva

Total tests: 201. Passed: 192. Failed: 0. Skipped: 9.
Test Run Successful.
Test execution time: 53.6460 Seconds

Task                          Duration            
--------------------------------------------------
Clean                         00:00:00.0271910    
Download-Binaries             Skipped             
Restore                       00:00:07.3961561    
Build                         00:00:04.1114974    
Test                          00:00:57.7240249    
--------------------------------------------------
Total:                        00:01:09.2588694    
```


## Debian (Stretch) - 2.2

_Configuration:_  
- .Net Core SDK - 2.2.105
- .Net Core Runtime - 2.2.3
- Debian - 9 (Stretch)

_Container setup:_  
```sh
docker run -it --rm -v $(pwd):/app mcr.microsoft.com/dotnet/core/sdk:2.2.105-stretch sh
```

_In the container:_  
```sh
cd /app
dotnet add tests/NetVips.Tests package NetVips.Native.linux-x64 -v 8.7.4-beta1
./build.sh --target=Test
```

_Test results:_  
```sh
 
```


## Debian (Stretch) - 3.0

_Configuration:_  
- .Net Core SDK - 3.0.100-preview3
- .Net Core Runtime - 3.0.0
- Debian - 9 (Stretch)

_Container setup:_  
```sh
docker run -it --rm -v $(pwd):/app mcr.microsoft.com/dotnet/core/sdk:3.0.100-preview3-stretch sh
```

_In the container:_  
```sh
cd /app
dotnet add tests/NetVips.Tests package NetVips.Native.linux-x64 -v 8.7.4-beta1
./build.sh --target=Test
```

_Test results:_  
```sh
 PENDING CAKE ISSUE
```


## ⏸ Alpine (3.8) - 2.2

_Configuration:_  
- .Net Core SDK - 2.2.105
- .Net Core Runtime - 2.2.3
- Alpine - 3.8

_Container setup:_  
```sh
docker run -it --rm -v $(pwd):/app mcr.microsoft.com/dotnet/core/sdk:2.2.105-alpine3.8 sh
```

_In the container:_  
```sh
cd /app
dotnet add tests/NetVips.Tests package NetVips.Native.linux-musl-x64 -v 8.7.4-beta1
export DOTNET_RUNTIME_ID=linux-musl-x64
./build.sh --target=Test
```

_Test results:_  
```sh
PENDING CAKE ISSUE
```


## ⏸ Alpine (3.9) - 2.2

_Configuration:_  
- .Net Core SDK - 2.2.105
- .Net Core Runtime - 2.2.3
- Alpine - 3.9

_Container setup:_  
```sh
docker run -it --rm -v $(pwd):/app mcr.microsoft.com/dotnet/core/sdk:2.2.105-alpine3.9 sh
```

_In the container:_  
```sh
cd /app
dotnet add tests/NetVips.Tests package NetVips.Native.linux-musl-x64 -v 8.7.4-beta1
export DOTNET_RUNTIME_ID=linux-musl-x64
./build.sh --target=Test
```

_Test results:_  
```sh
PENDING CAKE ISSUE
```


## ⏸ Alpine (3.9) - 3.0

_Configuration:_  
- .Net Core SDK - 3.0.100-preview3
- .Net Core Runtime - 3.0.0
- Alpine - 3.9

_Container setup:_  
```sh
docker run -it --rm -v $(pwd):/app mcr.microsoft.com/dotnet/core/sdk:3.0.100-preview3-alpine3.9 sh
```

_In the container:_  
```sh
cd /app
dotnet add tests/NetVips.Tests package NetVips.Native.linux-musl-x64 -v 8.7.4-beta1
export DOTNET_RUNTIME_ID=linux-musl-x64
./build.sh --target=Test
```

_Test results:_  
```sh
PENDING CAKE ISSUE
```

**Comment by sandcastle** [comment]
@kleisauke - Are you happy to set up Circle CI to test the Linux/Alpine distros for both .Net 2.2 and 3.0?

**Comment by kleisauke** [comment]
@sandcastle Thanks for testing this! I've just made some improvements with commit 180e2fc2c1bc67dfbe8043425200c2d4ba050639.

In that commit I've switched the build automation system to [Nuke](https://nuke.build/), this should fix the `unterminated quoted string` errors. I've also conditionally included the precompiled binaries in the test solution: https://github.com/kleisauke/net-vips/blob/180e2fc2c1bc67dfbe8043425200c2d4ba050639/tests/NetVips.Tests/NetVips.Tests.csproj#L27-L33

This will make testing easier, you can just run `./build.sh` (the `TestWithNuGetBinaries` will be set if it's not running on Travis CI).

The .NET Core 3 Preview SDK currently fails with:
```
It was not possible to find any compatible framework version
The specified framework 'Microsoft.NETCore.App', version '2.2.0' was not found.
  - Check application dependencies and target a framework version installed at:
      /usr/share/dotnet/
  - The .NET Core Runtime and SDK can be installed from:
      https://aka.ms/dotnet-download
  - The following versions are installed:
      3.0.0-preview3-27503-5 at [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  - Installing .NET Core prerequisites might help resolve this problem:
      https://go.microsoft.com/fwlink/?linkid=2063370
```

I could add the `netstandard2.1` to the `TargetFrameworks` property, but I think I've to wait with that because it is still in the preview phase. I'm not using C# 8.0 features, so it should be fine. All other Docker commands that are not dependent on the .NET Core 3 Preview SDK seem to work for me.

I'm not sure if using Circle CI offers any advantages. I already use Travis CI and AppVeyor to test the Linux, MacOS and Windows precompiled binaries.

**Comment by kleisauke** [comment]
Also, sorry for the delay of the 1.1.0 release. I've been busy with my studies and other opensource projects.

Perhaps I should release 1.1.0 with the pre-compiled binaries of libvips 8.8 (which will be released this or next month).

**Comment by sandcastle** [comment]
> Perhaps I should release 1.1.0 with the pre-compiled binaries of libvips 8.8 (which will be released this or next month).

Yes, I think it's worth associating 1.1.0 to the 8.8 release of libvips. If we can get some pre-release nuget packages up, I can run it in prod for a while against 1000's of images we have.

**Comment by sandcastle** [comment]
> I'm not sure if using Circle CI offers any advantages. I already use Travis CI and AppVeyor to test the Linux, MacOS and Windows precompiled binaries.

Sorry, ignore this comment, I didn't realise you were already using Travis.

**Comment by sandcastle** [comment]
Nuke tests all working well ✅, I have updated the above:
https://github.com/kleisauke/net-vips/issues/21#issuecomment-479683139

**Comment by kleisauke** [comment]
Quick update: I've just released the final version of the native binaries (version 8.7.4) on NuGet (relevant commit https://github.com/kleisauke/net-vips/commit/9b0a35ff36ea3a2f979c2ab6ce083c6b5a83e927). 

The OS X binaries includes PR lovell/package-libvips-darwin#3 (processing jpeg files on macOS should be quicker now). I've also published a new [`NetVips.Native`](https://www.nuget.org/packages/NetVips.Native/) package on NuGet which depends on the other `NetVips.Native.*` packages. This can be useful for cross-platform deployments, you only need to add one extra package.

I'll publish a pre-release of NetVips 1.1 when a release candidate of libvips 8.8 is released. The final NetVips 1.1 release will be in sync with the release of libvips 8.8.

**Comment by kleisauke** [comment]
I've just released NetVips 1.1.0-rc1 and NetVips.Native 8.8.0-rc1 on NuGet. If you would like to see what's new in libvips 8.8, please visit the the release notes of libvips:
https://libvips.github.io/libvips/2019/04/22/What's-new-in-8.8.html

**Comment by sandcastle** [comment]
@kleisauke I'm seeing some issues with one of my projects where its not finding `libvips.so.42`. 

Any ideas how to debug what is causing this?

```
System.Exception: Could not resize image. ---> 
System.DllNotFoundException: Unable to load shared library 'libvips.so.42' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: liblibvips.so.42.so: cannot open shared object file: No such file or directory
```

The project file has the following references:

```xml
<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup>
    <TargetFramework>netcoreapp2.2</TargetFramework>
    <OutputType>Exe</OutputType>
    <LangVersion>latest</LangVersion>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Aspose.PDF" Version="19.4.0" />
    <PackageReference Include="Aspose.Words" Version="19.4.0" />
    <PackageReference Include="Aspose.Cells" Version="19.4.0" />
    <PackageReference Include="Aspose.Email" Version="19.4.0" />
    <PackageReference Include="JetBrains.Annotations" Version="2019.1.1" />
    <PackageReference Include="Microsoft.AspNetCore" Version="2.2.0" />
    <PackageReference Include="Microsoft.AspNetCore.HttpsPolicy" Version="2.2.0" />
    <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="2.2.0" />
    <PackageReference Include="Google.Cloud.Diagnostics.AspNetCore" Version="2.0.1" />
    <PackageReference Include="Newtonsoft.Json" Version="12.0.2" />
    <PackageReference Include="Serilog" Version="2.8.0" />
    <PackageReference Include="Serilog.Sinks.Console" Version="3.1.1" />
    <PackageReference Include="Serilog.AspNetCore" Version="2.1.1" />
    <PackageReference Include="SkiaSharp" Version="1.68.0" />
    <PackageReference Include="SkiaSharp.NativeAssets.Linux" Version="1.68.0" />
    <PackageReference Include="NetVips" Version="1.1.0-rc2" />
    <PackageReference Include="NetVips.Native" Version="8.8.0-rc3" />
  </ItemGroup>

</Project>
```

It's running in a `mcr.microsoft.com/dotnet/core/aspnet:2.2` container:

```docker
FROM mcr.microsoft.com/dotnet/core/aspnet:2.2

ADD . /app
WORKDIR /app

ENV ASPNETCORE_ENVIRONMENT=Production

ENTRYPOINT ["dotnet", "App.Files.dll"]
```

I have checked that the files are correctly in the container application folder:

```sh
cd /app

ls | grep .so

libSkiaSharp.so
libcairo-gobject.so.2
libcairo.so.2
libclrjit.so
libcoreclr.so
libcoreclrtraceptprovider.so
libcroco-0.6.so.3
libdbgshim.so
libexif.so.12
libexpat.so.1
libffi.so.6
libfontconfig.so.1
libfreetype.so.6
libfribidi.so.0
libgdk_pixbuf-2.0.so.0
libgif.so.7
libgio-2.0.so.0
libglib-2.0.so.0
libgmodule-2.0.so.0
libgobject-2.0.so.0
libgrpc_csharp_ext.x64.so
libgrpc_csharp_ext.x86.so
libgsf-1.so.114
libgthread-2.0.so.0
libharfbuzz.so.0
libhostfxr.so
libhostpolicy.so
libjpeg.so.8
liblcms2.so.2
libmscordaccore.so
libmscordbi.so
liborc-0.4.so.0
libpango-1.0.so.0
libpangocairo-1.0.so.0
libpangoft2-1.0.so.0
libpixman-1.so.0
libpng16.so.16
librsvg-2.so.2
libsos.so
libsosplugin.so
libtiff.so.5
libuuid.so.1
libvips.so.42
libwebp.so.7
libwebpdemux.so.2
libwebpmux.so.3
libxml2.so.2
libz.so.1
```

Running `ldd` I get the following:

```sh
ldd libvips.so.42 

linux-vdso.so.1 (0x00007ffdb4dd6000)
libpng16.so.16 => /app/./libpng16.so.16 (0x00007fc19e091000)
libz.so.1 => /app/./libz.so.1 (0x00007fc19de76000)
libtiff.so.5 => /app/./libtiff.so.5 (0x00007fc19dbf3000)
libjpeg.so.8 => /app/./libjpeg.so.8 (0x00007fc19d95e000)
libgmodule-2.0.so.0 => /app/./libgmodule-2.0.so.0 (0x00007fc19d75b000)
libexpat.so.1 => /app/./libexpat.so.1 (0x00007fc19d52b000)
libgsf-1.so.114 => /app/./libgsf-1.so.114 (0x00007fc19d2e3000)
libxml2.so.2 => /app/./libxml2.so.2 (0x00007fc19cff5000)
liborc-0.4.so.0 => /app/./liborc-0.4.so.0 (0x00007fc19cd3e000)
liblcms2.so.2 => /app/./liblcms2.so.2 (0x00007fc19cacb000)
libgif.so.7 => /app/./libgif.so.7 (0x00007fc19c8c2000)
librsvg-2.so.2 => /app/./librsvg-2.so.2 (0x00007fc19bfc2000)
libgio-2.0.so.0 => /app/./libgio-2.0.so.0 (0x00007fc19bc08000)
libgdk_pixbuf-2.0.so.0 => /app/./libgdk_pixbuf-2.0.so.0 (0x00007fc19b9d7000)
libgobject-2.0.so.0 => /app/./libgobject-2.0.so.0 (0x00007fc19b782000)
libglib-2.0.so.0 => /app/./libglib-2.0.so.0 (0x00007fc19b427000)
libcairo.so.2 => /app/./libcairo.so.2 (0x00007fc19b10a000)
libwebpmux.so.3 => /app/./libwebpmux.so.3 (0x00007fc19aeff000)
libwebpdemux.so.2 => /app/./libwebpdemux.so.2 (0x00007fc19acfb000)
libwebp.so.7 => /app/./libwebp.so.7 (0x00007fc19aa65000)
libexif.so.12 => /app/./libexif.so.12 (0x00007fc19a820000)
libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007fc19a49e000)
libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fc19a19a000)
libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fc199dfb000)
libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fc199be4000)
libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fc1999c7000)
libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fc1997c3000)
librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007fc1995bb000)
libcairo-gobject.so.2 => /app/./libcairo-gobject.so.2 (0x00007fc1993b3000)
libresolv.so.2 => /lib/x86_64-linux-gnu/libresolv.so.2 (0x00007fc19919c000)
libpangocairo-1.0.so.0 => /app/./libpangocairo-1.0.so.0 (0x00007fc198f8f000)
libpixman-1.so.0 => /app/./libpixman-1.so.0 (0x00007fc198cbd000)
libpangoft2-1.0.so.0 => /app/./libpangoft2-1.0.so.0 (0x00007fc198aa7000)
libharfbuzz.so.0 => /app/./libharfbuzz.so.0 (0x00007fc1987a9000)
libpango-1.0.so.0 => /app/./libpango-1.0.so.0 (0x00007fc19855d000)
libgthread-2.0.so.0 => /app/./libgthread-2.0.so.0 (0x00007fc19835c000)
libfribidi.so.0 => /app/./libfribidi.so.0 (0x00007fc19813f000)
libffi.so.6 => /app/./libffi.so.6 (0x00007fc197f35000)
libfontconfig.so.1 => /app/./libfontconfig.so.1 (0x00007fc197ce2000)
libuuid.so.1 => /app/./libuuid.so.1 (0x00007fc197adc000)
libfreetype.so.6 => /app/./libfreetype.so.6 (0x00007fc197815000)
libcroco-0.6.so.3 => /app/./libcroco-0.6.so.3 (0x00007fc1975d5000)
/lib64/ld-linux-x86-64.so.2 (0x00007fc19ea9f000
```

I don't have a lot of experience debugging linux dependency issues, so any assistance would be greatly appreciated :)

**Comment by kleisauke** [comment]
Hi @sandcastle,

Are you able to provide a standalone project that exhibits this behavior? I couldn't reproduce this with the same dependencies. My Dockerfile:
```dockerfile
FROM mcr.microsoft.com/dotnet/core/sdk:2.2 AS build
WORKDIR /app

# Copy csproj and restore as distinct layers
COPY *.sln .
COPY *.csproj ./
RUN dotnet restore

# Copy everything else and build
COPY . ./
RUN dotnet publish -c Release -o out

# Build runtime image
FROM mcr.microsoft.com/dotnet/core/aspnet:2.2 AS runtime
WORKDIR /app

COPY --from=build /app/out .

ENV ASPNETCORE_ENVIRONMENT=Production

ENTRYPOINT ["dotnet", "net-vips-asp.dll"]
```

Example project:  [net-vips-asp.zip](https://github.com/kleisauke/net-vips/files/3232439/net-vips-asp.zip)

**Comment by kleisauke** [comment]
NetVips v1.1.0 / NetVips.Native v8.8.1 is now available.

Feel free to open a new issue if you're still experiencing problems with the pre-compiled libvips binaries.
